### PR TITLE
Complete Override Manifest for react-native-win32

### PR DESCRIPTION
**Pull request by NickGerleman**
This change fills out the parts of the override manifest that tooling couldn't figure out. Quite a few overrides no longer exist upstream, or are no longer needed, so we get to remove them here. There are a couple of other trivial changes to remove some overrides.

We make liberal use of the "derived" type here, even for simple stubs. This means that our stubs will follow Facebook stubs, and we will get a lint break (once validation is enabled) if Facebook removes files. This is seemingly valuable, given how many of our stubs don't track anything upstream.

Validated that we can bundle, build, linting passes, and that override validation now passes.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4205)

### CODE REVIEW

File: packages/react-native-win32/src/Libraries/Components/StatusBar/StatusBar.win32.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
 /**
- * This copies what StatusBar.android.js does
+ * Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (NickGerleman) [inline packages/react-native-win32/src/Libraries/Components/StatusBar/StatusBar.win32.ts]: The comment on the left isn't accurate.

An aside, we're missing copyright headers on a lot of these files. From previous discussions it seems like that isn't a major deal, but we should probably address that.

File: packages/react-native-win32/src/Libraries/Image/ImageTypes.ts [hunk 2]
```diff
@@ -1,9 +1,15 @@
-import { ViewProps } from 'react-native';
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (NickGerleman) [inline packages/react-native-win32/src/Libraries/Image/ImageTypes.ts]: This got lumped in. Can revert if wanted.

File: packages/react-native-win32/src/Libraries/Network/RCTNetworking.win32.ts [hunk 3]
```diff
@@ -1,11 +1,101 @@
...
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
  */
 'use strict';
 
 // Do not require the native RCTNetworking module directly! Use this wrapper module instead.
 // It will add the necessary requestId, so that you don't have to generate it yourself.
 
-import { RCTNetworking } from './Network';
```
Comment (NickGerleman) [inline packages/react-native-win32/src/Libraries/Network/RCTNetworking.win32.ts]: Network isn't directly included anywhere. Moved thw RCTNetworking code back to the main file for consistency with other platforms.

File: packages/react-native-win32/src/Libraries/Components/StatusBar/StatusBar.win32.ts (hunk 1 above)
Comment (NikoAri) [reply packages/react-native-win32/src/Libraries/Components/StatusBar/StatusBar.win32.ts]: If we want to release it in Open Source, full header should be:

// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

---
In reply to: [385375547](https://github.com/microsoft/react-native-windows/pull/4205#discussion_r385375547) [](ancestors = 385375547)

File: packages/react-native-win32/src/Libraries/Components/StatusBar/StatusBar.win32.ts [hunk 4]
```diff
@@ -1,5 +1,10 @@
...
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
```
Comment (tkamireh) [inline packages/react-native-win32/src/Libraries/Components/StatusBar/StatusBar.win32.ts]: Consider removing this file or using an `unimplementedView`

File: packages/react-native-win32/src/overrides.json [hunk 5]
```diff
@@ -177,75 +149,59 @@
...
-      "issue": "LEGACY_FIXME"
+      "type": "platform",
+      "file": "Libraries\\Components\\TextInput\\Tests\\TextInputTest.tsx"
     },
     {
-      "type": "???",
+      "type": "derived",
       "file": "Libraries\\Components\\TextInput\\TextInput.Types.win32.ts",
-      "baseFile": "???",
+      "baseFile": "Libraries\\Components\\TextInput\\TextInput.js",
```
Comment (tkamireh) [inline packages/react-native-win32/src/overrides.json]: let's make this "`type": "platform"` and remove the basefile

**Pull request merged**